CMakeLists.txt

```diff
@@ -113,16 +114,52 @@ elseif(MLX_BUILD_METAL)
   target_link_libraries(mlx PUBLIC ${METAL_LIB} ${FOUNDATION_LIB} ${QUARTZ_LIB})
 endif()
 
+if(WIN32)
+  if(MSVC)
```

Comment:
I'm just curious why we check both of these. Is it common to use a compiler other than MSVC on windows?

Reply:
Some people use MinGW (https://github.com/JuliaPackaging/Yggdrasil/pull/9761), which provides layers for some POSIX APIs.